src/robotFactory.js

```diff
@@ -1,10 +1,101 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    { x = 0, y = 0 },
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
```

Comment:
This will make it more readable and understandable.

```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```